migrate-about-us.rb

```diff
@@ -93,21 +98,111 @@
+    panitia_or_admin_names.include?(name)
+  end
+  missing = about_us_names.select do |name|
+    !panitia_or_admin_names.include?(name)
+  end
+  return existing, missing
+end
 
-// Randomize the order!
-function randomize_array(arr) {
-  var current_index = arr.length;
+def process_alumni(panitia_or_admin)
+  alumni_names = $alumni_data.map do |alumni|
+    alumni[:name].downcase
+  end
+  panitia_or_admin_names = panitia_or_admin.map do |user|
+    user.fullname.downcase
+  end
+  existing = alumni_names.select do |name|
+    panitia_or_admin_names.include?(name)
+  end
+  missing = alumni_names.select do |name|
+    !panitia_or_admin_names.include?(name)
+  end
+  return existing, missing
+end
 
-  while (current_index !== 0) {
-    var random_index = Math.floor(Math.random() * current_index);
-    current_index--;
+def create_missing_users(missing)
+  missing.map do |name|
+    words = name.split
+    first_word = words[0]
+    second_word = words[1]
+    new_user = User.create(
+      :username => "#{first_word}#{second_word}",
+      :email => "#{first_word}#{second_word}@gmail.com",
+      :password => "#{first_word}#{second_word}",
+      :fullname => name,
+      :school => "NUS",
+      :province_id => 12,
+      :status_id => 8
+    )
+    new_user.enable
+    new_user.add_role :panitia
+    new_user.add_role :admin
+  end
+end
 
-    var temporary_value = arr[current_index];
-    arr[current_index] = arr[random_index];
-    arr[random_index] = temporary_value;
-  }
-}
+def create_about_users(panitia_or_admin)
+  $about_us_data.map do |about_us|
+    image_path = about_us[:image]
+    name = about_us[:name]
+    description = about_us[:description]
+    image = File.open(Rails.root.join('app', 'assets', 'images', 'panitia', image_path), 'r')
+    related_account = panitia_or_admin.select { |user| user.fullname.downcase == name.downcase }.first
+    related_account.about_user = AboutUser.create(
+      :name => name,
+      :description => description,
+      :image => image,
+      :is_alumni => false
+    )
+    related_account.save
+  end
+  $alumni_data.map do |alumni|
+    image_path = alumni[:image]
+    name = alumni[:name]
+    description = alumni[:description]
+    image = File.open(Rails.root.join('app', 'assets', 'images', 'panitia', image_path), 'r')
+    related_account = panitia_or_admin.select { |user| user.fullname.downcase == name.downcase }.first
+    related_account.about_user = AboutUser.create(
+      :name => name,
+      :description => description,
+      :image => image,
+      :is_alumni => true
+    )
+    related_account.save
+  end
+end
 
-randomize_array(about_us_data);
-randomize_array(alumni_data);
+def migrate
+  panitia_or_admin = get_panitia_or_admin()
+  existing_panitia_or_admin, missing_panitia_or_admin = process_about_us(panitia_or_admin)
+  existing_alumni, missing_alumni = process_alumni(panitia_or_admin)
+  create_missing_users(missing_panitia_or_admin)
```

Comment:
Instead of creating missing users with rather arbitrary defaults, let us just print those users that are missing. We can act on them separately (maybe the name is a typo somewhere, etc).

Reply:
I see, so we'll create only for those that already exist then? (and perhaps manually ask for those missing to create on their own)

Reply:
Essentially yeah. My concern with us creating the users is that I feel like there are no defaults that can apply to everyone, and if in the end we need to change them anyway, might as well us create them.